src/components/StakingPage/index.tsx

```diff
@@ -79,6 +81,34 @@ const Staking = () => {
     return true;
   }, [chainId]);  
 
+  const sortedPosition = useMemo(() => {
+    let archivedPosition = [];
+    let unarchivePosition = [];
+    
+    vaults.forEach((vault, index) => {
+      if(vault?.result?.hakkaAmount.isZero()) {
+        archivedPosition.push({...vault, 'index': index});
+      } else {
+        unarchivePosition.push({...vault, 'index': index});
+      }
+    });
+    
```

Comment:
archivedPosition is no relation to line 97-110
I think could separate to two steps, count origin [unarchivePosition, archivedPosition] first